File: search/binary_search.py
def main():
    list =  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    target = int(input("Target: "))
    index = binary_search(list, target, 0, len(list) - 1)
    print(f'Target Index: {index}')



def binary_search(list, target, first, last):
    midpoint = (last + first) // 2
    print(f"Current Midpoint Index: {midpoint}\nCurrent Midpoint: {list[midpoint]} Target: {target}")

    if first > last:
        return "Not Found"
    elif list[midpoint] == target:
        return midpoint
    elif list[midpoint] < target:
        return binary_search(list, target, midpoint + 1, last)
    else:
        return binary_search(list, target, first, midpoint - 1)


if __name__ == "__main__":
    main()

File: search/binary_search.c
#include <stdio.h>
#include <string.h>

int binary_search(int list[], int target, int first, int last);
int main() {
    int list[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int index = binary_search(list, 10, 0, sizeof(list)/sizeof(int) - 1);
    printf("%i\n", index);
}


int binary_search(int list[], int target, int first, int last){
    int midpoint = (last + first) / 2;
    printf("Current Mid-point: %i\n", midpoint);
    if(list[midpoint] == target){
        return midpoint;
    }
    else if(list[midpoint] < target){
        return binary_search(list, target, midpoint + 1, last);
    }
    else{
       return binary_search(list, target, first, midpoint - 1);
    }
}